From ddbd744e08e6823bf1757a65e7967e83fab87190 Mon Sep 17 00:00:00 2001
From: Sultan Alsawaf <sultan@kerneltoast.com>
Date: Tue, 3 May 2022 23:21:44 -0700
Subject: [PATCH] PM / freezer: Reduce freeze timeout to 1 second for Android

Freezing processes on Android usually takes less than 100 ms, and if it
takes longer than that to the point where the 20 second freeze timeout is
reached, it's because the remaining processes to be frozen are deadlocked
waiting for something from a process which is already frozen. There's no
point in burning power trying to freeze for that long, so reduce the freeze
timeout to a very generous 1 second for Android and don't let anything mess
with it.

Signed-off-by: Sultan Alsawaf <sultan@kerneltoast.com>
---
 kernel/power/main.c    | 3 +++
 kernel/power/process.c | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

Index: common/kernel/power/main.c
===================================================================
--- common/kernel/power/main.c	2025-10-07 00:24:01.861724178 +1030
+++ common/kernel/power/main.c	2025-10-07 00:24:01.851724178 +1030
@@ -921,6 +921,9 @@
 {
 	unsigned long val;
 
+	/* Don't let anything in Android change the freeze timeout */
+	return n;
+
 	if (kstrtoul(buf, 10, &val))
 		return -EINVAL;
 
Index: common/kernel/power/process.c
===================================================================
--- common/kernel/power/process.c	2025-10-07 00:24:01.861724178 +1030
+++ common/kernel/power/process.c	2025-10-07 00:24:01.851724178 +1030
@@ -25,7 +25,7 @@
 /*
  * Timeout for stopping processes
  */
-unsigned int __read_mostly freeze_timeout_msecs = 20 * MSEC_PER_SEC;
+unsigned int __read_mostly freeze_timeout_msecs = MSEC_PER_SEC;
 
 static int try_to_freeze_tasks(bool user_only)
 {
